### PR TITLE
Disable Features based on CSV annotations

### PR DESCRIPTION
**Pull request by bipuladh**
- Disables Storage System List Page
- Disables Add Capacity Kebab Item
- Sets flags based on ocs-operator CSV
Screenshot: 
Screenshot from 2021-12-15 12-40-42
Related PR: 
https://github.com/red-hat-storage/odf-console/pull/75

### CODE REVIEW

**Comment by bipuladh** [comment]
Reference ODF PR: https://github.com/red-hat-storage/ocs-operator/pull/1425

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 1]
```diff
@@ -52,20 +53,25 @@ export enum GUARDED_FEATURES {
...
+  [FEATURES.OCS_MULTUS]: 'multus',
+  [FEATURES.OCS_ARBITER]: 'arbiter',
+  [FEATURES.OCS_KMS]: 'kms',
+  [FEATURES.OCS_FLEXIBLE_SCALING]: 'flexible-scaling',
+  [FEATURES.OCS_TAINT_NODES]: 'taint-nodes',
+  [FEATURES.OCS_THICK_PROVISION]: 'thick-provision',
+  [FEATURES.OCS_POOL_MANAGEMENT]: 'pool-management',
+  [FEATURES.OCS_NAMESPACE_STORE]: 'namespace-store',
+  [FEATURES.ODF_MCG_STANDALONE]: 'mcg-standalone',
+};
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/features.ts:70]:
```suggestion
[FEATURES.ODF_HPCS_KMS]: 'hpcs-kms',
};
```

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/features.ts:70]: `[FEATURES.OCS_KMS]: 'kms'` is for Vault

File: frontend/packages/ceph-storage-plugin/src/features.ts [hunk 2]
```diff
@@ -52,20 +53,25 @@ export enum GUARDED_FEATURES {
   OCS_NAMESPACE_STORE = 'OCS_NAMESPACE_STORE',
   ODF_MCG_STANDALONE = 'ODF_MCG_STANDALONE',
   ODF_HPCS_KMS = 'ODF_HPCS_KMS',
+  SS_LIST = 'SS_LIST',
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/features.ts:56]: Isn't add-capacity modal also migrated in the same PR ?

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/features.ts:56]: ack, missed out.

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 3]
```diff
@@ -504,6 +505,9 @@ const plugin: Plugin<ConsumedExtensions> = [
           )
         ).default,
     },
+    flags: {
+      disallowed: [FEATURES.SS_LIST],
+    },
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:510]: don't we want to disallow add-capacity extension as well in this plugin file ??

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:510]: yeah just saw it, don't know why we have it in both the places though.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:510]: IIRC, this one is for add-capacity kebab on `Installed Operators> Operator details` page.

**Comment by SanjalKatiyar** [comment]
tested... list page was hidden when it was disabled from CSV.
Screenshot from 2021-12-15 17-00-40
Screenshot from 2021-12-15 17-01-19

**Pull request merged**